zap/src/main/java/org/parosproxy/paros/core/scanner/AbstractPlugin.java

```diff
@@ -323,6 +326,29 @@ protected void sendAndReceive(
         parent.performScannerHookAfterScan(message, this);
     }
 
+    /**
+     * Updates the Content-Length header of the request.
+     *
+     * <p>For methods with no nor expected enclosed content the header is removed otherwise in all
+     * other cases the header is updated to match the length of the content.
+     *
+     * @param message the message to update.
+     * @since 2.12.0
+     */
+    protected void updateRequestContentLength(HttpMessage message) {
+        int bodyLength = message.getRequestBody().length();
+        String method = message.getRequestHeader().getMethod();
+        if (bodyLength == 0
+                && (HttpRequestHeader.CONNECT.equals(method)
+                        || HttpRequestHeader.DELETE.equals(method)
+                        || HttpRequestHeader.GET.equals(method)
+                        || HttpRequestHeader.HEAD.equals(method))) {
```

Comment:
Should include OPTIONS?

Reply:
I didn’t check the RFCs

Reply:
The OPTIONS can have content, should add TRACE though.